packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -122,9 +141,43 @@ export class SchemaConverter implements SimpleDbSchemaConverter {
       p = p.next(() => this.ensureSequenceNumbers(txn));
     }
 
+    assert(
+      fromVersion <= 8,
+      'After version 8, we switch to a separately maintained schema ' +
+        'version. IndexedDB should never see its version go past 8.'
+    );
+    if (fromVersion < 8 && toVersion >= 8) {
+      p = p.next(() => addManualSchemaVersion(txn, db));
+    }
+
+    // IMPORTANT: new migrations should be implemented in
+    // `upgradeFromManualVersion`. After this point, we no longer use
+    // IndexedDB's versioning, as it does not allow downgrades.
+
     return p;
   }
 
+  doManualMigrations(
+    db: IDBDatabase,
+    toVersion: number
+  ): PersistencePromise<void> {
+    return readManualSchemaVersion(db).next(fromVersion => {
+      // We don't yet know what migrations will run, and we don't know which ones might be skipped
+      // that added object stores that we will need to access. Rather than attempt to pick the exact
+      // right set of object stores based on the current version, add all of the existing object
+      // stores to the transaction.
```

Comment:
FWIW I think you could drop this comment if you wanted.  It seems pretty reasonable to me to just lock all the stores when we're doing migrations. I don't think it requires explanation.